src/govuk/i18n.mjs

```diff
@@ -15,19 +15,56 @@ function I18n (translations) {
  * returns the appropriate string.
  *
  * @param    {String}  lookupKey  - The lookup key of the string to use.
+ * @param    {Object}  options    - Any options passed with the translation string, e.g: for string interpolation.
  * @returns  {String}             - The appropriate translation string.
  */
-I18n.prototype.t = function (lookupKey) {
+I18n.prototype.t = function (lookupKey, options) {
   if (!lookupKey) {
     // Print a console error if no lookup key has been provided
     throw new Error('i18n lookup key missing.')
   } else if (lookupKey in this.translations) {
-    // Return the translation string if it exists in the object
-    return this.translations[lookupKey]
+    // Fetch the translation string for that lookup key
+    var translationString = this.translations[lookupKey]
+
+    // Check for ${} placeholders in the translation string
+    if (translationString.match(/\${(.\S+)}/)) {
+      if (!options) {
+        throw new Error('i18n: cannot replace placeholders in string if no option data provided')
+      }
+
+      return this.replacePlaceholders(translationString, options)
+    } else {
+      return translationString
+    }
   } else {
     // Otherwise, return the lookup key itself as the fallback
     return lookupKey
   }
 }
 
+/**
+ * Takes a translation string with placeholders, and replaces the placeholders
+ * with the provided data
+ *
+ * @param    {String}  translationString  - The translation string
+ * @param    {Object}  options    - Any options passed with the translation string, e.g: for string interpolation.
+ * @returns  {String}             - The translation string to output, with ${} placeholders replaced
+ */
+I18n.prototype.replacePlaceholders = function (translationString, options) {
+  var placeholderRegex = RegExp(/\${(.\S+)}/, 'g')
+  var placeholderMatch
+
+  // Use `exec` for fetching regex matches, as matchAll() is not supported in IE
+  while ((placeholderMatch = placeholderRegex.exec(translationString)) !== null) {
+    var placeholderKey = placeholderMatch[1]
+    if (!options[placeholderKey]) {
+      throw new Error('i18n: no data found to replace ' + placeholderMatch[0] + ' placeholder in string')
+    } else {
+      translationString = translationString.replace(placeholderMatch[0], options[placeholderKey])
```

Comment:
If you're not familiar with the output of `RegExp.match` then it's not 100% clear what this is doing.

Worth assigning `placeholderMatch[0]` to a variable so we can give it a useful name like we have with `placeholderKey`?

Reply:
Yep... any opinions on the naming? I'm thinking `placeholderMatchString`, `placeholderToReplace` - something that's enough to differentiate it from the placeholderKey 🤔

Reply:
Maybe something like `placeholderWithDelimiters`, `placeholderWithBraces` or `placeholderIncludingBraces`?